srcalc/src/main/java/gov/va/med/srcalc/vista/vistalink/VistaLinkProcedureCaller.java

```diff
@@ -75,30 +84,55 @@ public VistaLinkVistaDao(final String division)
         }
     }
     
-    public VistaPerson loadVistaPerson(final String duz)
+    @Override
+    public String getDivision()
     {
-        fLogger.debug("Loading VistaPerson for duz {}.", duz);
+        return fDivision;
+    }
 
+    @Override
+    public List<String> doRpc(final String duz, final String rpcName, final String... args)
+    {
         // This is the RPC context for all ASRC RPCs. (This value is determined
         // by VistA.)
         final String rpcContext = "SR ASRC";
 
         final VistaLinkDuzConnectionSpec cs =
                 new VistaLinkDuzConnectionSpec(fDivision, duz);
-        
+
         try
         {
+            fLogger.debug("About to call remote procedure \"{}\"", rpcName);
             final RpcRequest req = RpcRequestFactory.getRpcRequest(
-                    rpcContext, "SR ASRC USER");
+                    rpcContext, rpcName);
+            
+            // Set the arguments.
+            for (int i = 0; i < args.length; ++i)
+            {
+                // VistA starts counting at 1, not 0.
+                final int vistaParamIndex = i + 1;
+                final String arg = args[i];
+                fLogger.debug("Setting parameter {} to {}", vistaParamIndex, arg);
+                req.getParams().setParam(vistaParamIndex, "string", arg);
+            }
+
             final VistaLinkConnection conn = (VistaLinkConnection)fVlcf.getConnection(cs);
             try
             {
                 final RpcResponse response = conn.executeRPC(req);
-                
-                // Get the first index in the array. We don't care about the rest.
-                final String userString = response.getResults().split("\n")[0];
-                fLogger.debug("Got user: ", userString);
-                return new VistaPerson(duz, userString, "user class not pulled");
+                fLogger.debug(
+                        "Got {} response: {}",
+                        response.getResultsType(), response.getResults());
+                if (RESULT_TYPE_ARRAY.equals(response.getResultsType()))
+                {
+                    // NB: String.split() strips trailing empty strings.
+                    return Arrays.asList(response.getResults().split("\n"));
```

Comment:
Should this be all white space? If you split on new lines and there are multiple new lines then you will end up with multiple empty strings. Maybe using [\n]+ would be better.